src/OpenFeature.Contrib.Hooks.Otel/MetricsHook.cs

```diff
@@ -22,12 +22,16 @@ public class MetricsHook : Hook
         private readonly Counter<long> _evaluationRequestCounter;
         private readonly Counter<long> _evaluationSuccessCounter;
         private readonly Counter<long> _evaluationErrorCounter;
+        private readonly KeyValuePair<string, object>[] _customDimensionsTagList;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="MetricsHook"/> class.
         /// </summary>
-        public MetricsHook()
+        /// <param name="customDimensions">The optional custom dimensions.</param>
+        public MetricsHook(MetricHookCustomDimensions customDimensions = null)
         {
+            _customDimensionsTagList = customDimensions?.GetTagList() ?? Array.Empty<KeyValuePair<string, object>>();
+
```

Comment:
The flexibility of this approach is a bit limited. I think a functional approach might be better. Instead of allowing users to define k/v pairs, I would recommend you allow users to define a function that takes an FlagEvaluationDetails object as a param, and returns a tuple or k/v set that will be added. Then users can build whatever data they want and add it as metics.

We do something similar in JS: https://github.com/open-feature/js-sdk-contrib/blob/main/libs/hooks/open-telemetry/README.md#custom-attributes

Reply:
@toddbaert I was looking into the Java implementation and they offer a Flag Metadata (that is not implemented on dotnet SDK yet) and uses a custom dimensions that have a predefined list. Might be interesting to keep the extra dictionary and when we have the flag metadata implemented at SDK level, we use the FlagEvaluationDetails metadata?

Reply:
Sorry I missed this comment for so long @askpt . I'm not sure I fully understand what you're proposing. Maybe you can clarify.

In any case, I think that once you implement the metadata [here](https://github.com/open-feature/dotnet-sdk/pull/223), the functional approach used in Java and JS should work.